cms-contentful/src/main/java/io/commercetools/sunrise/cms/contentful/ContentfulCmsService.java

```diff
@@ -1,17 +1,95 @@
 package io.commercetools.sunrise.cms.contentful;
 
+import com.contentful.java.cda.CDAAsset;
+import com.contentful.java.cda.CDAClient;
+import com.contentful.java.cda.CDAEntry;
+import com.contentful.java.cda.SynchronizedSpace;
 import io.commercetools.sunrise.cms.CmsIdentifier;
 import io.commercetools.sunrise.cms.CmsService;
 
 import java.util.List;
 import java.util.Locale;
+import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
+import java.util.function.Predicate;
 
 public class ContentfulCmsService implements CmsService {
 
+    private Map<String, CDAEntry> entries;
+
+    ContentfulCmsService(Map<String, CDAEntry> entries) {
+        this.entries = entries;
+    }
+
     @Override
     public CompletionStage<Optional<String>> get(final List<Locale> locales, final CmsIdentifier cmsIdentifier) {
-        return null; // TODO
+        return CompletableFuture.supplyAsync(() -> getEntry(cmsIdentifier)
+                .map(entry -> getLocalizedField(locales, entry, cmsIdentifier.getFieldName()))
+                .orElseGet(Optional::empty));
+    }
+
+    // TODO create models
+    private Optional<CDAEntry> getEntry(CmsIdentifier cmsIdentifier) {
+        return entries.values()
```

Comment:
If the CMS has a lot of entries this lookup is expensive. 

In case of the fully cached CMS thing, a map based structure should be used for fast lookups.